### ISSUE TITLE
when use tensorflow to predict images with different resolution,the forward's cost  unstable

### ISSUE THREAD

**Issue by kasyoukin** [tensorflow/tensorflow#22369]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Linux Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:binary
- **TensorFlow version (use command below)**:1.10.1
- **Python version**:3.5
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:9.0/7.1
- **GPU model and memory**:Tesla P4 ,8GB
- **Exact command to reproduce**:no

### Describe the problem
when use tensorflow to predict images with different resolution,the forward's cost  unstable. my image long side is less than 1280 , the resize code below:
```
 def resize_im(self, im, scale=600, max_scale=1280): 
      f = 1.0
      if scale > min(im.shape[0], im.shape[1]): 
        f = float(scale) / min(im.shape[0], im.shape[1]) 
      if max_scale != None and f * max(im.shape[0], im.shape[1]) > max_scale: 
        f = float(max_scale) / max(im.shape[0], im.shape[1]) 
      return cv2.resize(im, None, None, fx=f, fy=f, interpolation=cv2.INTER_LINEAR), f 
```

[use different resolution.txt](https://github.com/tensorflow/tensorflow/files/2395362/use.different.resolution.txt)
see the net_fw

when padding image to 1280*1280, the forward's cost stable,between 0.25 and 0.3,but the small image will cost more.

I think it is BFC‘s problem.
How can I solve this problem?I think if tensorflow can lock the chunk's memory (use input image with resolution 1280*1280) for all layers , this problem could be sovled

**Comment by kasyoukin** [comment]
anyone can response me?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ymodak: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yuefengz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yuefengz** [comment]
@reedwm Reed, do you have any idea?

**Comment by reedwm** [comment]
@kasyoukin, what do you mean by "the forward's cost"? I am not quite sure what the issue is.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @reedwm: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
Hi @kasyoukin ! You should normalize your image resolution in the dataset for optimum results. It can be done through[ image generators ](https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator)through the [target_size](https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator#:~:text=Default%3A%20None.-,target_size,-tuple%20of%20integers) (Target length and width)parameter . It also has a data augmentation feature which can give better results during validation. Thank you!
